File: jiwon/두큐합같게만들기.java
import java.util.*;
import java.util.stream.Stream;
import java.util.stream.Collectors;
class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int answer = -2;
        long sum1=0; long sum2=0;

        Queue<Integer> q1= new LinkedList<>(Arrays.stream(queue1).boxed().collect(Collectors.toList()));
        Queue<Integer> q2= new LinkedList<>(Arrays.stream(queue2).boxed().collect(Collectors.toList()));



        for( int i : queue1) sum1+= (long)i;
        for( int i : queue2) sum2+= (long)i;

        int time=0;
        while( time < 10000000){ //처음 큐 길이의 3배 정도만 되도 괜찮다고 한다.
            if(sum1==sum2) {
                answer=time;
                break;
            }
            time++;
            if(sum1 > sum2){

                int out = q1.poll();
                q2.add(out);
                sum1 -= out;
                sum2 += out;
            }else{

                int out = q2.poll();
                q1.add(out);
                sum2 -= out;
                sum1 += out;
            }
        }

        // System.out.println(Collections.unmodifiableList(q1));
        // System.out.println(sum1 + " " + sum2);
        if(answer==-2) return -1;
        else return answer;
    }
}

File: joon/src/programmers/kakao2022internship/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    #     적은 큐에서 큰 큐로 이동
    #     반복을 중단 하는 경우 : 같아질때, 전체 큐 크기의 3배 돌렸을떄?

    q1 = deque(queue1)
    q2 = deque(queue2)

    s1 = sum(q1)
    s2 = sum(q2)

    index = 0

    if (s1 + s2) % 2 == 1:
        return -1

    while len(queue1) * 4 > index:
        if s1 > s2:
            temp = q1.popleft()
            q2.append(temp)
            s1 -= temp
            s2 += temp
        elif s2 > s1:
            temp = q2.popleft()
            q1.append(temp)
            s2 -= temp
            s1 += temp
        else:
            return index

        index += 1

    return -1